### ISSUE TITLE
Matrix2 get scale incorrect

### ISSUE THREAD

**Issue by leftnoodle** [craftworkgames/MonoGame.Extended#587]
It returns incorrect values and doesn't take into account negative scale values. This can be seen when creating a matrix using CreateFrom with rotation = 0.296706 radians, scaleX =7 and scaleY = 4. The result from get scale is 6.796, 4.338.

I believe the correct calculation should be:

var scaleX = (float)Math.Sign(M11) * (float)Math.Sqrt(M11 * M11 + M12 * M12);
var scaleY = (float)Math.Sign(M22) * (float)Math.Sqrt(M21 * M21 + M22 * M22);

The get rotation also returns incorrect rotation values

**Comment by craftworkgames** [comment]
Thanks for reporting the issue.

To be honest, I didn't write most of that code and I don't use it very often. So it's difficult to tell just from looking at your description what the problem is or if that's the correct solution.

To make this change I suggest we need some unit tests and / or find a good reference for the equations.

**Comment by lithiumtoast** [comment]
I wrote the code. 

@leftnoodle You are correct it does not take into account negative scale values. My proposed solution.

```
public void Decompose(out Vector2 scale, out float rotation, out Vector2 translation)
{
     translation.X = this.M31;
     translation.Y = this.M32;

     float xs = (Math.Sign(M11 * M12) < 0) ? -1 : 1;
     float ys = (Math.Sign(M21 * M22) < 0) ? -1 : 1;

     scale.X = xs * (float)Math.Sqrt(M11 * M11 + M12 * M12);
     scale.Y = ys * (float)Math.Sqrt(M21 * M21 + M22 * M22);

     rotation = (float)Math.Atan2(M21, M11);
}
```

For two dimensions, the code can be chopped to get methods (properties) as the status-quo. However, for three dimensions [it's been discussed why this is not ideal](https://github.com/MonoGame/MonoGame/issues/5553). To keep consistency with MonoGame I propose marking the get properties, such as `Scale`, deprecated and rely on the new `Decompose` method. The similar pattern can be found with [`Matrix`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Matrix.cs#L1454).

**Comment by leftnoodle** [comment]
I think you will need to do the same as Matrix.cs and divide M21 and M11 by scale.Y and scale.X respectively to obtain the correct rotation?

I'm also not sure that will calculate the correct scale, I could be mistaken but if we take a standard rotation matrix with PI / 4.0 radians then we get:

M11 = 0.707, M12 = 0.707  xs = 1
M21 = -0.707, M22 = 0.707 ys = -1

Which would result in a negative scale.Y? I might be completely wrong though as  I'm doing this in my head.

**Comment by lithiumtoast** [comment]
@leftnoodle 

The problem is that there is a [historical inconsistency between notation of two-dimensional mathematics in academia and computer graphics](https://gamedev.stackexchange.com/questions/83570/why-is-the-origin-in-computer-graphics-coordinates-at-the-top-left).

**Comment by lithiumtoast** [comment]
In v4 we will consider using [`Matrix3x2`](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.matrix3x2) instead.

**Comment by AristurtleDev** [comment]
This is resolved in #870 with the introduction of `Matrix3x2`
